pandas/indexes/base.py

```diff
@@ -2936,6 +2936,9 @@ def _wrap_joined_index(self, joined, other):
         name = self.name if self.name == other.name else None
         return Index(joined, name=name)
 
+    def _get_string_slice(self, key, use_lhs=True, use_rhs=True):
```

Comment:
Can you add a docstring or comment explaining where this is overwritten / used ?